server_tests/test/clientTests.js

```diff
@@ -1138,27 +1155,29 @@ describe("RUST API TEST SUITE", function() {
                         Comparison.bublPollOutput(res, true, {'id': commandId, 'fingerprint': fingerprint});
                         fingerprint = res.body.fingerprint;
                         return Q.delay(4000);
-                    }, shouldNotGetHere)
+                    })
                     .then( function() {
-                        return testClient.bublPoll(commandId, fingerprint, 4);
+                        return testClient.bublPoll(commandId, fingerprint, 5);
                     })
                     .then( function onSuccess (res) {
                         Comparison.bublPollOutput(res, false, {'id': commandId, 'fingerprint': fingerprint});
                         return testClient.bublStop(commandId);
-                    }, shouldNotGetHere)
+                    })
                     .then( function onSuccess (res) {
                         Comparison.bublStopOutput(res);
-                    }, shouldNotGetHere);
+                        deferred.resolve();
+                    })
+                    .catch(deferred.reject);
                 }
-            })
+            }), deferred.promise])
             .then( function onSuccess (res) {
-                Comparison.bublCaptureVideoOutput(res);
+                Comparison.bublCaptureVideoOutput(res[0]);
```

Comment:
What's going on here, why is there an array? (if it's because of the `all`, you probably want to move this `then` to be applied directly to the `captureVideo` promise)